src/NLog.Web/NLogRequestPostedBodyHttpModule.cs

```diff
@@ -0,0 +1,169 @@
+    /// __nlog-aspnet-request-posted-body
+    ///
+    /// To use, subclass this class and set your own Configuration
+    /// </summary>
+    public class NLogRequestPostedBodyHttpModule : IHttpModule
+    {
+        /// <summary>
+        /// The name of the HttpModule
+        /// You may override in the subclass with your own name
+        /// </summary>
+        public string ModuleName { get; set; } = "NLog Request Posted Body Module";
+
+        /// <summary>
+        /// The configuration for the HttpModule
+        /// </summary>
+        public NLogRequestPostedBodyMiddlewareConfiguration Configuration { get; set; } =
+            NLogRequestPostedBodyMiddlewareConfiguration.Default;
+
+
+        /// <summary>
+        /// Hook in to the BeginRequest event to capture the request posted body
+        /// </summary>
+        /// <param name="context"></param>
+        public void Init(HttpApplication context)
+        {
+            context.BeginRequest += InterceptRequest;
+        }
+
+        /// <summary>
+        /// This will forward the necessary arguments to the capture request body method
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="args"></param>
+        protected void InterceptRequest(object sender, EventArgs args)
+        {
+            HttpApplication app = sender as HttpApplication;
+
+            CaptureRequestPostedBody(
+                app?.Request?.InputStream,
+                app?.Context?.Items,
+                Configuration.ShouldCapture(app));
+        }
+
+        /// <summary>
+        /// Public to be unit testable, HttpContext and HttpRequest are un-mockable
+        /// unless you are using ASP.NET Core.  HttpContext and HttpRequest are sealed
+        /// and no not have an interface so NSubstitute throws an Exception mocking them.
+        /// </summary>
+        /// <param name="bodyStream"></param>
+        /// <param name="items"></param>
+        /// <param name="shouldCapture"></param>
+        public void CaptureRequestPostedBody(
+            Stream bodyStream,
+            IDictionary items,
+            bool shouldCapture)
+        {
+            if (bodyStream == null)
+            {
+                InternalLogger.Debug("NLogRequestPostedBodyMiddleware: HttpContext.Request.InputStream stream is null");
+                return;
+            }
+
+            if (!bodyStream.CanRead)
+            {
+                InternalLogger.Debug("NLogRequestPostedBodyMiddleware: HttpContext.Request.InputStream stream is non-readable");
+                return;
+            }
+
+            if (!bodyStream.CanSeek)
+            {
+                InternalLogger.Debug("NLogRequestPostedBodyMiddleware: HttpContext.Request.InputStream stream is non-seekable");
+                return;
+            }
+
+            if (shouldCapture)
+            {
+                items[AspNetRequestPostedBodyLayoutRenderer.NLogPostedRequestBodyKey] = GetString(bodyStream);
```

Comment:
When `GetString(bodyStream)` returns `null`  or `string.Empty`, then it should not insert into items

Reply:
Completed as requested